joomla/joomla-cms
# Don't allow installing a library which overwrites core folders

Pull Request for Issue #23868.

### Summary of Changes
Adding a list of folders to the installer library adapter which are restricted from installing.
This prevents library extensions to overwrite core files.

### Testing Instructions
@mbabker created a test extension which replaces the libraries/vendor/autoload.php file:
https://www.babdev.com/downloads/destructor.zip
Of course installing this breaks Joomla, so use with care and be prepared to restore the file and manually remove the extension in the database.
See original issue for more context.


### Expected result
Libraries are not allowed to overwrite core folders.

### Actual result
Libraries with any name can be installed, potentially bringing down your Joomla.



### Documentation Changes Required
None
